### PR TITLE
 #169258663 Get Info About The Most Traveled Destination

### PR DESCRIPTION
**Pull request by william000000**
#### What does this PR do?
 User should be able to get information about the most traveled destination
#### Description of Task to be completed?
- Get information about the most traveled destination
- Add tests 
- Swagger documentation 
#### How should this be manually tested?
- You have to clone the repo `git clone https://github.com/andela/team-odd-bn-backend.git`
- Then After the repo cloned `cd project-folder` and run `npm i` to install all dependencies
- Go to my branch `git checkout ft-get-most-traveled-destination-169258663`
- Insert the data by `npx sequelize-cli db:seed:all`
- Run app by `npm run dev-start`

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#169258663](https://www.pivotaltracker.com/story/show/169258663)
#### Screenshots (if appropriate)
image

image

#### Questions:
N/A

### CODE REVIEW

File: src/helpers/TripHelper.js [hunk 1]
```diff
@@ -40,6 +42,39 @@ class TripHelper {
...
+    }
+    const maxNumber = Math.max(...countHolder);
+    const frequentCityId = newArray.filter((item) => {
+      if (item.counter === maxNumber) {
+        return { cityId: item.cityId, counts: item.counter };
+      }
+    });
+    const [{ cityId, counter }] = frequentCityId;
+    return { cityId, counter };
+  }
```
Comment (hezronkimutai) [inline src/helpers/TripHelper.js:97]: Wow! I guess you spend a lot of time  to come up with such a beautiful algorithm. Good work.

File: src/routes/api/tripRoute.js [hunk 2]
```diff
@@ -421,4 +429,32 @@ tripRouter
...
+ *          description: No token provided
+ *        "404":
+ *          description: No Trip found
+ */
+
+tripRouter
+  .get(
+    '/mostTraveled', verifyToken, isUserVerified, mostTraveledDestination
+  );
+
```
Comment (hezronkimutai) [inline src/routes/api/tripRoute.js:522]: @william000000  Though this format is also valid, I prefer you to use the below format 
```
tripRouter
  .get(
    '/mostTraveled', 
    verifyToken, 
    isUserVerified,
    mostTraveledDestination
  );
```

**Review by hezronkimutai — CHANGES_REQUESTED** [review]
@william000000 , Good work, I left some comments, please view it.

File: src/helpers/TripHelper.js (hunk 1 above)
Comment (william000000) [reply src/helpers/TripHelper.js:97]: You are right @hezronkimutai, Thank you so much for that acknowledgment!

**Review by hezronkimutai — APPROVED** [review]

File: src/routes/api/tripRoute.js [hunk 3]
```diff
@@ -421,4 +429,35 @@ tripRouter
     operateAcceptOrReject, acceptOrRejectRequestsController
   );
 
+/**
+ * @swagger
+ *
+ * /trips/mostTraveled:
+ *    get:
```
Comment (NiyoEric) [inline src/routes/api/tripRoute.js]: I would like to request to change the API route name. I recommend using `-` hyphen to improve the readability of API

Screen Shot 2019-12-12 at 14 27 00

More info are [here](https://restfulapi.net/resource-naming/)

**Review by NiyoEric — CHANGES_REQUESTED** [review]
Well done @william000000. You delivered the best work. I have left one feedback. Please take a look.

File: src/routes/api/tripRoute.js (hunk 3 above)
Comment (william000000) [reply src/routes/api/tripRoute.js]: Thanks for the new skill you shared with me, I have changed it now, you can take a look!

**Review by NiyoEric — APPROVED** [review]

File: src/helpers/TripHelper.js [hunk 4]
```diff
@@ -61,6 +62,39 @@ class TripHelper {
...
+      }
+      countHolder.push(count);
+      newArray.push({ counter: count, cityId: getDestinationIds[i] });
+    }
+    const maxNumber = Math.max(...countHolder);
+    const frequentCityId = newArray.filter((item) => {
+      if (item.counter === maxNumber) {
+        return { cityId: item.cityId, counts: item.counter };
+      }
+    });
```
Comment (victkarangwa) [inline src/helpers/TripHelper.js:94]: Noce shot @william000000

File: src/routes/api/tripRoute.js [hunk 5]
```diff
@@ -480,7 +486,40 @@ tripRouter
...
+ *          required: true
+ *          schema:
+ *            type: string
+ *      responses:
+ *        "200":
+ *          description:  Most traveled destination info retrieved successfully
+ *        "401":
+ *          description: No token provided
+ *        "404":
+ *          description: No Trip found
```
Comment (victkarangwa) [inline src/routes/api/tripRoute.js]: Do we really need this?
I mean this 404 error?

**Review by victkarangwa — COMMENTED** [review]
Good job @william000000 
Just attend to the feedback I left

**Review by victkarangwa — APPROVED** [review]

File: src/routes/api/tripRoute.js (hunk 5 above)
Comment (william000000) [reply src/routes/api/tripRoute.js]: I removed it @victorkarangwa4

File: src/helpers/TripHelper.js (hunk 4 above)
Comment (william000000) [reply src/helpers/TripHelper.js:94]: Thanks @victorkarangwa4

**Review by julietezekwe — APPROVED** [review]

**Pull request merged**